### PR TITLE
Replace Artist Project test with new Blockly Project test using migrated lab

### PR DESCRIPTION
**Pull request by mikeharv**
The test in `artist_project.feature` fails in Google Blockly as it is written due to differences between APIs. To handle other tests like this, we will often add `?blocklyVerison=google` to the test URL and then update the steps as needed. This way, we can begin testing on Google Blockly immediately, and keep the actual flip PR minimal.
Unfortunately, this test doesn't work with that strategy due to multiple changes to the active URL. When redirecting to the URL with channel id, and then to the share URL, the blockly URL parameter would be lost.

I can't find any evidence that this test needs to use Artist, specifically. It was originally impremented 9 years ago to test projects generally: https://github.com/code-dot-org/code-dot-org/blob/b6846a381b1349c4c4756220d0009db1eb341e25/dashboard/test/ui/features/projects.feature
We now have other tests for projects with labs like App Lab and Game Lab. If we reconsider this test's purpose as being for Blockly projects, it shouldn't matter which specific lab we use. 

This PR renames `artist_project.feature` to `blockly_project.feature` and then updates it test to use Dance, which has been migrated already.

Like many other UI tests, this one relies on block IDs, so I updated the project level to give a block id to the start block (`setup`). Previously, the test used two steps to 1) open a toolbox category and 2) drag it to the start block. Since this test isn't about those specific interactions, I went for a more direct route. Now, we simply 1) add a new block to the workspace and 2) connect it to the start block. The latter step definition already existed. No simulated user actions are involved.
Because Dance Party prompts for user age, I also needed to add a step to handle that.

## Links
https://codedotorg.atlassian.net/browse/CT-737

## Testing story
See PR description

## Follow-up work

If there was a reason to use Artist for this particular test, we could of course change it again after the migration is complete, but that doesn't feel necessary in my option.

\
## PR Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: dashboard/config/levels/custom/dance/New Dance Lab Project.level [hunk 1]
```diff
@@ -48,23 +48,23 @@
     "mini_rubric": "false",
     "validation_enabled": "false",
     "hide_pause_button": "false",
+    "blockly_variables": "dancer1",
```
Comment (molly-moen) [inline dashboard/config/levels/custom/dance/New Dance Lab Project.level:51]: is there any risk with adding these new fields? I'm guessing they are new defaults, but we want to be careful editing the new project level.

File: dashboard/test/ui/features/step_definitions/blockly.rb [hunk 2]
```diff
@@ -60,11 +71,6 @@
   @browser.execute_script code
 end
 
-When /^I drag block matching selector "([^"]*)" to block matching selector "([^"]*)"$/ do |from, to|
```
Comment (molly-moen) [inline dashboard/test/ui/features/step_definitions/blockly.rb:63]: Is this step definition (and the one in the other file) no longer used anywhere?

**Review by molly-moen — COMMENTED** [review]
Couple questions, looks good overall!

File: dashboard/config/levels/custom/dance/New Dance Lab Project.level (hunk 1 above)
Comment (mikeharv) [reply dashboard/config/levels/custom/dance/New Dance Lab Project.level:51]: This field, and some others that look new are actually just in new positions. See line 55 (old version) below.

File: dashboard/test/ui/features/step_definitions/blockly.rb (hunk 2 above)
Comment (mikeharv) [reply dashboard/test/ui/features/step_definitions/blockly.rb:63]: That's correct! Sorry, I should have mentioned that in the PR description.

File: dashboard/config/levels/custom/dance/New Dance Lab Project.level (hunk 1 above)
Comment (molly-moen) [reply dashboard/config/levels/custom/dance/New Dance Lab Project.level:51]: oh thanks! reviewing too fast 😅

**Review by molly-moen — APPROVED** [review]

**Pull request merged**